src/App.jsx

```diff
@@ -15,12 +15,37 @@ function App() {
   function handleDeleteTodo(id) {
     setTodos(todos.filter((todo) => todo.id !== id));
   }
+  function handleToggleTodo(id) {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      )
+    );
+  }
+  function handleEdit(id, newContent) {
+    setTodos(
+      todos.map((todo) =>
+        id === todo.id ? { ...todo, content: newContent } : todo
+      )
+    );
+  }
+
+  function handleSetCurrentTodoId(id) {
+    setCurrentEditTodoId(id);
+  }
 
   return (
     <div>
       <h1>To Do List</h1>
       <TodoForm onSubmit={handleAddTodo} />
-      <TodoList todos={todos} onDelete={handleDeleteTodo} />
+      <TodoList
+        todos={todos}
+        onDelete={handleDeleteTodo}
+        onToggle={handleToggleTodo}
+        onEdit={handleEdit}
+        currentEditTodoId={currentEditTodoId}
+        onSetCurrentTodoId={handleSetCurrentTodoId}
```

Comment:
The props of `onEdit` and `onSetCurrentTodoId` do not make sense and can easily cause confusion:
1. `onEdit` is more like an event when the todo edit button is clicked, not the submit.
2. `onSetCurrentTodoId` covers both events: edit button click and cancel button click. We should not have the `TodoList` understand the `TodoApp` state; they should be fully decoupled.

I suggest using a more common props definition:
1. `onEdit`: when the TodoItem Edit Button is clicked.
2. `onSubmit`: when the TodoItem Submit Button is clicked.
3. `onCancel`: when the TodoItem Cancel Button is clicked.